### PR TITLE
Our verify scripts should fail the pipeline if any of our commands fail

### PR DESCRIPTION
**Pull request by tyler-ball**
### Description

Our pipeline is passing `chef-run version` when it should not be.

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

Signed-off-by: tyler-ball <tyleraball@gmail.com>

### CODE REVIEW

File: ci/verify.bat [hunk 1]
```diff
@@ -5,7 +5,16 @@ REM ; package is working correctly after install.
...
 REM ; chef-run version ensures our bin ends up on path and the basic ruby env is
 REM ; working.
-call chef-run version
+chef-run --version || goto :error
 
 REM ; Ensure our ChefDK works
 chef env
+
+REM ; Test
+chef-run version || goto :error
```
Comment (tyler-ball) [inline ci/verify.bat]: Running a build through with this intentional failure right now to make sure this pattern works on Windows

Comment (tyler-ball) [reply ci/verify.bat]: The unix tests are passing an ad-hoc run well, but Windows is printing the following:
```
chef-run: 0.1.136
Regular expression run condition: Expression=[^(?!windows).*$], Label=[windows-2012r2]
Run condition [Regular expression match] preventing perform for step [Execute shell]
```

It correctly prints the version from `chef-run --version` on line 8 above. But this line should be failing and it is not. I tried running a similar batch file on my Windows workstation and it seemed to fail appropriately. @tduffield or @scotthain any ideas?

Comment (tduffield) [reply ci/verify.bat]: I do not unfortunately. If I find some time I can do some research, but maybe someone more familiar with Windows might know?

Comment (scotthain) [reply ci/verify.bat]: Try an iteration of `chef-run --version || echo "Exited with error %errorlevel%" && exit /b` and the like!

**Pull request merged**